packages/trip-form/src/MetroModeSelector/utils.ts

```diff
@@ -0,0 +1,311 @@
+import {
+  ModeButtonDefinition,
+  ModeSetting,
+  ModeSettingValues,
+  TransportMode
+} from "@opentripplanner/types";
+import { useState } from "react";
+import { QueryParamConfig, decodeQueryParams } from "serialize-query-params";
+import {
+  useQueryParam,
+  DelimitedArrayParam,
+  ObjectParam
+} from "use-query-params";
+
+import coreUtils from "@opentripplanner/core-utils";
+import { QueryParamChangeEvent } from "../types";
+
+const { queryGen } = coreUtils;
+const { TRANSIT_SUBMODES_AND_TRANSIT } = queryGen;
+
+type InitialStateType = {
+  enabledModeButtons: string[];
+  modeSettingValues: ModeSettingValues;
+};
+
+type ModeStateConfig = {
+  queryParamState?: boolean;
+};
+
+/**
+ * Aggregates all the modes from the input mode button definitions
+ * Should probably filter unique values, but it's not possible with a Set due to them being objects
+ * @param modeButtonDefinitions Array of mode buttons
+ * @returns All the (unique) modes from the buttons
+ */
+export function aggregateModes(
+  modeButtonDefinitions: ModeButtonDefinition[]
+): TransportMode[] {
+  return modeButtonDefinitions.reduce<Array<TransportMode>>((array, combo) => {
+    combo.modes.forEach(mode => array.push(mode));
+    return array;
+  }, new Array<TransportMode>());
+}
+
+/**
+ * Filters mode buttons by list of keys, used to find enabled buttons.
+ * TODO: Remove this function? Is it needed?
+ * @param modeButtonDefinitions All mode definitions
+ * @param keys List of keys of buttons to include
+ * @returns Filtered list of buttons
+ */
+export function filterModeDefitionsByKey(
+  modeButtonDefinitions: ModeButtonDefinition[],
+  keys: string[]
+): ModeButtonDefinition[] {
+  return modeButtonDefinitions.filter(def => keys.includes(def.key));
+}
+
+/**
+ * Connects the mode setting values from a values object, where each key corresponds
+ * to a mode setting in the modeSettings parameter.
+ * @param modeSettings The mode settings with empty `value` params
+ * @param values An object containing setting values
+ * @returns Mode settings with values populated
+ */
+export function populateSettingsWithValues(
+  modeSettings: ModeSetting[],
+  values: ModeSettingValues
+): ModeSetting[] {
+  return modeSettings.map(setting => {
+    const value = values[setting.key];
+    let convertedVal;
+    if (setting.type === "CHECKBOX") {
+      // If the parameter is anything besides the string "true", it will be false
+      convertedVal = value === "true" || value === true;
+    } else if (setting.type === "SLIDER") {
+      convertedVal = Number(value);
+    }
```

Comment:
This might be cleaner as a method that converts `setting.type` into a converted value, then that can bet set to a const